test/quantization/pt2e/test_metadata_porting.py

```diff
@@ -0,0 +1,384 @@
+    ]
+)
+
+
+class TestMetaDataPorting(QuantizationTestCase):
+    def _test_metadata_porting(
+        self,
+        model,
+        example_inputs,
+        quantizer,
+        node_tags=None,
+    ):
+        m_eager = model.eval()
+
+        # program capture
+        m = copy.deepcopy(m_eager)
+        m = export.capture_pre_autograd_graph(
+            m,
+            example_inputs,
+        )
+
+        m = prepare_pt2e(m, quantizer)
+        # Calibrate
+        m(*example_inputs)
+        m = convert_pt2e(m)
+
+        pt2_quant_output = m(*example_inputs)
+        recorded_node_tags = {}
+        for n in m.graph.nodes:
+            if (
+                n.op == "call_function"
+                and n.target in _QUANT_OPS
+                and "quantization_tag" in n.meta
+            ):
+                if n.target not in recorded_node_tags:
+                    recorded_node_tags[n.target] = set()
+                if n.meta["quantization_tag"] in recorded_node_tags[n.target]:
+                    raise ValueError(
+                        f"{n} has tag {n.meta['quantization_tag']} that is associated with another node of the same type"
+                    )
+                recorded_node_tags[n.target].add(n.meta["quantization_tag"])
+        self.assertEqual(set(recorded_node_tags.keys()), set(node_tags.keys()))
+        for k, v in recorded_node_tags.items():
+            self.assertEqual(v, node_tags[k])
+
+    def test_simple_metadata_porting(self):
+        """
+        Model under test
+        conv2d -> avgpool -> hardtanh -> linear
+        Check quantization tags on conv2d, avgpool and linear are correctly set
+        """
+
+        class BackendAQuantizer(Quantizer):
+            def annotate(self, gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+                backend_string = "BackendA"
+                quantization_config = get_symmetric_quantization_config(
+                    is_per_channel=True
+                )
+                annotated_partitions = OP_TO_ANNOTATOR["linear"](
+                    gm, quantization_config
+                )
+                _tag_partitions(backend_string, "linear", annotated_partitions)
+                annotated_partitions = OP_TO_ANNOTATOR["conv2d"](
+                    gm, quantization_config
+                )
+                _tag_partitions(backend_string, "conv2d", annotated_partitions)
+                annotated_partitions = OP_TO_ANNOTATOR["adaptive_avg_pool2d"](
+                    gm, quantization_config
+                )
+                _tag_partitions(
+                    backend_string, "adaptive_avg_pool2d", annotated_partitions
+                )
+
+            def validate(self, model: torch.fx.GraphModule) -> None:
+                pass
+
+            @classmethod
+            def get_supported_operators(cls) -> List[OperatorConfig]:
+                pass
+
+        example_inputs = (torch.randn(1, 3, 5, 5),)
+        quantizer_per_tensor_tags = set(
```

Comment:
nit: quantize_per_tensor_tags? same for `dequantizer_per_tensor_tags` and `dequantizer_per_channel_tags`